c2rust-transpile/src/translator/variadic.rs

```diff
@@ -9,130 +9,110 @@ pub enum VaPart {
 
 macro_rules! match_or {
     ([$e:expr] $p:pat => $r:tt) => {
-        let $r = match $e {
-            $p => $r,
-            _ => return None,
-        };
+        match $e {
+            $p => Some($r),
+            _ => None,
+        }
     };
     ([$e:expr] $p:pat if $g:expr => $r:tt) => {
-        let $r = match $e {
-            $p if $g => $r,
-            _ => return None,
-        };
+        match $e {
+            $p if $g => Some($r),
+            _ => None,
+        }
     };
 }
 
 impl<'c> Translation<'c> {
-    /// Returns true iff `va_start`, `va_end`, or `va_copy` may be called on `decl_id`.
+    /// Returns `true` iff `va_start`, `va_end`, or `va_copy` may be called on `decl_id`.
     pub fn is_va_decl(&self, decl_id: CDeclId) -> bool {
-        let fn_ctx = self.function_context.borrow();
-        if let Some(ref decls) = fn_ctx.va_list_decl_ids {
-            decls.contains(&decl_id)
-        } else {
-            false
-        }
+        self.function_context
+            .borrow()
+            .va_list_decl_ids
+            .as_ref()
+            .map(|decls| decls.contains(&decl_id))
+            .unwrap_or_default()
```

Comment:
`Default` for `bool` returning false is pretty arbitrary and relying on it is less clear here than `.unwrap_or(false)` would be.

Reply:
Fixed in https://github.com/immunant/c2rust/pull/638/commits/eaccb5cfa2440f79eb0c350f34a40b92b6ad0346.

Reply:
I would prefer you revert commit 3866a89 altogether. The existing code was good enough and your rewritten code needed improving (I had the same thought as Frances that using `.unwap_or_default()` is less clear so). Please avoid code churn where the effort to review it and the risk of introducing bugs doesn't outweigh the benefits.

Reply:
Okay, sorry about that, I can revert those two commits.

Reply:
Reverted in https://github.com/immunant/c2rust/pull/638/commits/1a5085a40e5f7f61a510869a8a069fab14a0c36f.